GSoC-2025/movement_support for Kalman filters_(Shraddha Bharadwaj).md

```diff
@@ -0,0 +1,125 @@
+
+
+## Project title
+
+movement: support for Kalman filters (Shraddha Bharadwaj)
+
+## Personal details
+Please include the following information:
+- **Full name** Shraddha Bharadwaj
+- **Email** shraddha5bharadwaj@gmail.com
+- **GitHub username** shraddha5bharadwaj
+- **Zulip username** https://neuroinformatics.zulipchat.com/#user/896544
+- **Location & time-zone** Atlanta, Georgia, United States, EST
+- **Personal website/project portfolio** https://github.com/shraddha5bharadwaj
+
+
+## Project proposal 
+
+
+- **Synopsis**
+
+     - The aim of the project is to implement a mathematical model such as the Kalman Filter to smooth trajectories using movement metadata and to implement accurate trajectory estimation for each animal concerned in trajectory tracking. 
+
+     - It very often happens that the way the behavior data is processed can vary the inference, I propose a workflow-based wrapper where the user can choose how the data will be processed and the model.
+
+        - Kalman Filter for trajectory prediction of a single animal, options of adapted Kalman Filters with features to accept latent features to be fed to the model. Compare the Adaptive Kalman Filter and variations of the same model.
+        - Create a data aggregator that can use all the input data and feed it to a single adapted Kalman filter that is essentially trained on all movements but can predict smoothened trajectory for a single animal. Allow for latent factor Modeling.
+         
+        - Run a customized trajectory prediction for each animal in consideration by feeding in the context of other animals as input to the model and infer trajectories for all animals from a stacked set of filters. 
+        - I propose we can bias the initialization of the filters for a single animal in a multi-animal environment with the position of other animals. A form of latent factor modeling. 
+    - The final deliverable would be a callable module where you can choose how the data will be treated, the latent factors that can shape the model, and the type of model that will be used for prediction. Allows the user more flexibility and an opportunity to truly experiment with the data.
+    
+
+- **Implementation timeline**
```

Comment:
How many hours are you planning to commit in total?

A project can be one of three sizes: small (90 h), medium (175 h) or large large (350 h).

It would also help to specify how many hours you have available per week (roughly).